File: sdk/typescript/src/api/system.ts
import type { HttpTransport } from '../transport.js';
import type { SystemStatus } from '../types.js';

export class SystemApi {
  constructor(private readonly http: HttpTransport) {}

  async version(): Promise<string> {
    const data = await this.http.execute<{ rockboxVersion: string }>(/* GraphQL */ `
      query Version { rockboxVersion }
    `);
    return data.rockboxVersion;
  }

  async status(): Promise<SystemStatus> {
    const data = await this.http.execute<{ globalStatus: SystemStatus }>(/* GraphQL */ `
      query GlobalStatus {
        globalStatus {
          resumeIndex resumeCrc32 resumeElapsed resumeOffset
          runtime topruntime dircacheSize
          lastScreen viewerIconCount lastVolumeChange
        }
      }
    `);
    return data.globalStatus;
  }
}

File: sdk/python/src/rockbox_sdk/api/system.py
"""System-level info: version, runtime status."""

from __future__ import annotations

from ..transport import HttpTransport
from ..types import SystemStatus


class SystemApi:
    def __init__(self, http: HttpTransport) -> None:
        self._http = http

    async def version(self) -> str:
        data = await self._http.execute("query Version { rockboxVersion }")
        return str(data["rockboxVersion"])

    async def status(self) -> SystemStatus:
        data = await self._http.execute(
            "query GlobalStatus { globalStatus { "
            "resumeIndex resumeCrc32 resumeElapsed resumeOffset "
            "runtime topruntime dircacheSize "
            "lastScreen viewerIconCount lastVolumeChange } }"
        )
        return SystemStatus.model_validate(data["globalStatus"])
